opentelemetry-sdk/tests/trace/test_sampling.py

```diff
@@ -537,3 +541,624 @@ def implicit_parent_context(span: trace.Span):
+        self.assertEqual(sampler.get_description(), 'RateLimitingSampler{0.1}')
+
+        # Test update
+        sampler = sampling.RateLimitingSampler(3.0)
+        sampler.rate_limiter.balance = 3.0
+        ts = time.time()
+        sampler.rate_limiter.last_tick = ts
+        with mock.patch('opentelemetry.sdk.trace.rate_limiter.RateLimiter.timestamp') \
+                as mock_time:
+            mock_time.side_effect = lambda: ts  # always return same time
+            self.assertEqual(sampler.rate_limiter.timestamp(), ts)
+            sampled = sampler.should_sample(None, 0, "span name").decision.is_sampled()
+            self.assertTrue(sampled)
+            self.assertEqual(sampler.rate_limiter.balance, 2.0)
+            self.assertEqual(sampler.get_description(), 'RateLimitingSampler{3.0}')
+
+            sampler.update(3.0)
+            self.assertEqual(sampler.get_description(), \
+                'RateLimitingSampler{3.0}', 'should short cirtcuit if rate is the same')
+
+            sampler.update(2.0)
+            self.assertEqual(sampler.rate_limiter.balance, 4.0 / 3.0)
+            self.assertEqual(sampler.get_description(), 'RateLimitingSampler{2.0}')
+
+    def test_guaranteed_throughput_probabilistic_sampler(self):
+        sampler = sampling.GuaranteedThroughputProbabilisticSampler('op', 2, 0.5)
+        sampler.lower_bound_sampler.rate_limiter.balance = 2.0
+        sample_result = sampler.should_sample(None, MAX_INT - 10, "span name")
+        self.assertTrue(sample_result.decision.is_sampled())
+        self.assertEqual(sample_result.attributes, {'sampler.type': 'traceidratio', 'sampler.param': 0.5})
+        sample_result = sampler.should_sample(None, MAX_INT + 10, "span name")
+        self.assertTrue(sample_result.decision.is_sampled())
+        self.assertEqual(sample_result.attributes, {'sampler.type': 'lowerbound', 'sampler.param': 0.5})
+        sample_result = sampler.should_sample(None, MAX_INT + 10, "span name")
+        self.assertFalse(sample_result.decision.is_sampled())
+        self.assertEqual(sampler.get_description(), 'GuaranteedThroughputProbabilisticSampler{op, 0.5, 2}')
+
+        sampler.update(3, 0.51)
+        sampler.lower_bound_sampler.rate_limiter.balance = 3.0
+        sample_result = sampler.should_sample(None, MAX_INT - 10, "span name")
+        self.assertTrue(sample_result.decision.is_sampled())
+        self.assertEqual(sample_result.attributes, {'sampler.type': 'traceidratio', 'sampler.param': 0.51})
+        sample_result = sampler.should_sample(None, int(MAX_INT + (MAX_INT / 4)), "span name")
+        self.assertTrue(sample_result.decision.is_sampled())
+        self.assertEqual(sample_result.attributes, {'sampler.type': 'lowerbound', 'sampler.param': 0.51})
+
+        self.assertEqual(sampler.get_description(), 'GuaranteedThroughputProbabilisticSampler{op, 0.51, 3}')
+
+    def test_sampler_equality(self):
+        const1 = sampling.StaticSampler(True)
```

Comment:
@srikanthccv for testing purposes in the ported code, these core samplers are meant to have equality implementations. Are we ok with me submitting equality tests and impl's into this repo?